### ISSUE TITLE
Using "IdS.MR", "Javascript Implicit Client", and "SampleWebApi" together| ClaimsIdentity has no claims

### ISSUE THREAD

**Issue by kalanbates** [thinktecture/Thinktecture.IdentityServer.v3.Samples#55]
TL;DR
How do I get IdentityServer's Claims available on the ClaimsPrincipal?
I can see Claims getting added to the User Profile Endpoint. I can see the "sub" come through as a requestedClaimType. I then see that the `(ClaimsPrincipal)principal.Identities.First().Claims` collection is empty. I was expecting it to contain a "sub" entry.

---
I'm having difficulty loading claims for an authenticated and authorized user.
I'm working from the IdS.MR implementation for my Identity System unchanged.

I also created a custom AuthorizeAttribute that (currently) simply checks that the Authorization Header exists on the incoming Request. I did this because the default Authorize attribute was always returning a 401.

I can confirm that the MembershipRebootUserService's `GetClaimsFromAccount(TAccount account)` is being called upon acceptance of the consent screen and is building a claims collection with 11 items.

I can also confirm that the Startup of the SampleWebApi is unchanged.
` JwtSecurityTokenHandler.InboundClaimTypeMap = new Dictionary<string, string>();`

But whenever my Authorized user accesses IdentityController's Get, principal.Identities.First().Claims contains nothing.  

Is there a mapping missing that I have to implement?
How do I get IdentityServer's Claims available on the ClaimsPrincipal?

**Comment by kalanbates** [comment]
Meh...thinking through the problem in a way to prepare it for asking the question has helped me answer the question; I need to decode the token during Authorization and pull the claims out of it. The claim doesn't exist on my ClaimsPrincipal because I have to put it there manually.
